components/UserDropdown.tsx

```diff
@@ -27,18 +27,15 @@ import { useTheme } from 'emotion-theming';
 import defaultTheme from './theme';
 import { Avatar, ChevronDown } from './theme/icons';
 import useAuthContext from '../global/hooks/useAuthContext';
-import { UserWithId } from '../global/types';
-import { InternalLink as Link } from './Link';
 import { useRouter } from 'next/router';
 import { USER_PATH } from '../global/utils/constants';
+import { WalletUser } from '../global/types';
 
-const getDisplayName = (user?: UserWithId) => {
+const getDisplayName = (user?: WalletUser) => {
   const greeting = 'Hello';
   if (user) {
-    if (user.firstName) {
-      return `${greeting}, ${user.firstName}`;
-    } else if (user.lastName) {
-      return `${greeting}, ${user.lastName}`;
+    if (user.name.length) {
```

Comment:
Just in case name goes poof for any reason.

```suggestion
    if (user.name?.length) {
```

Reply:
If there is a chance for a user property to be missing we should have the WalletUser type have this as an optional property, then TS will catch this stuff.

If we get the WalletUser from an API, then we need to do one of two things:
A) assume all props are optional
B) Write a schema validator to ensure the response is correct and throw an error if we get the wrong value from the API, then type convert the response of type `any` to `WalletUser` only once we know all the props are there.

Reply:
The user here will be provided from outside this codebase, and we shouldn't safely assume it'll always be available, even if we make it mandatory. 
Typescript wouldn't catch such an issue during runtime, and so I'd rather not get a `cannot find property 'length' of undefined` party stopper; even if TypeScript helps me code easier.